demos/slack-alert-app/go/slackbot.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+	"github.com/slack-go/slack"
+	"go.withpixie.dev/pixie/src/api/go/pxapi"
+	"go.withpixie.dev/pixie/src/api/go/pxapi/types"
+)
+
+var (
+	// Slack channel for Slackbot to post in.
+	// Slack App must be a member of this channel.
+	slackChannel = "#pixie-alerts"
+
+	// This PxL script ouputs a table of the HTTP total requests count and
+	// HTTP error (>4xxx) count for each service in the `px-sock-shop` namespace.
+	pxlScript = `
+import px
+
+df = px.DataFrame(table='http_events', start_time='-5m')
+
+# Add column for HTTP response status errors.
+df.error = df.http_resp_status >= 400
+
+# Add columns for service, namespace info
+df.namespace = df.ctx['namespace']
+df.service = df.ctx['service']
+
+# Filter for px-sock-shop namespace only.
+df = df[df.namespace == 'px-sock-shop']
+
+# Group HTTP events by service, counting errors and total HTTP events.
+df = df.groupby(['service']).agg(
+    error_count=('error', px.sum),
+    total_requests=('http_resp_status', px.count)
+)
+
+px.display(df, "http_table")
+`
+)
+
+func main() {
+
+	// The slackbot requires the following configs, which are specified
+	// using environment variables. For directions on how to find these
+	// config values, see: https://docs.pixielabs.ai/tutorials/slackbot-alert
+	pixieApiKey, ok := os.LookupEnv("PIXIE_API_KEY")
+	if !ok {
+		panic("Please set PIXIE_API_KEY environment variable.")
+	}
+
+	pixieClusterID, ok := os.LookupEnv("PIXIE_CLUSTER_ID")
+	if !ok {
+		panic("Please set PIXIE_CLUSTER_ID environment variable.")
+	}
+
+    slackToken, ok := os.LookupEnv("SLACK_BOT_TOKEN")
+	if !ok {
+		panic("Please set SLACK_BOT_TOKEN environment variable.")
+	}
+
+	ctx := context.Background()
+	pixieClient, err := pxapi.NewClient(ctx, pxapi.WithAPIKey(pixieApiKey))
+	if err != nil {
+		panic(err)
+	}
+	vz, err := pixieClient.NewVizierClient(ctx, pixieClusterID)
+	if err != nil {
+		panic(err)
+	}
+
+	slackClient := slack.New(slackToken)
+
+	// Send tick every 5 minutes.
```

Comment:
"Run the script every 5 min"?

Reply:
// Using a ticker (rather than a for loop + sleep call) means that the slackbot
// will message every 5 minutes, regardless of how long it takes to call the 
// Pixie API (as long as it takes less than 5 minutes).

Reply:
Maybe just 
// Send result every 5 min 
?